LoopKitUI/Extensions/OrientationLock.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  OrientationLock.swift
+//  LoopKitUI
+//
+//  Created by Michael Pangburn on 7/28/20.
+//  Copyright © 2020 LoopKit Authors. All rights reserved.
+//
+
+import UIKit
+
+/// To enable orientation locking via OrientationLock, the application's delegate must conform to OrientationLockable.
+public protocol OrientationLockable: AnyObject {
+    var supportedInterfaceOrientations: UIInterfaceOrientationMask { get set }
+}
+
+/// A class whose lifetime defines the application's supported interface orientations.
+///
+/// Construct an OrientationLock as a `@State` property in a SwiftUI view to constrain the view's supported orientations.
+/// To function, `OrientationLock.deviceOrientationController` must be assigned prior to use.
+public final class OrientationLock {
+    private let originalSupportedInterfaceOrientations: UIInterfaceOrientationMask
+
+    /// The global controller for device orientation.
+    /// The property must be assigned prior to instantiating any OrientationLock.
+    public static weak var deviceOrientationController: OrientationLockable!
```

Comment:
In Loop, this hook is a reference to the app delegate, which can control the app's supported orientations globallyl.

Reply:
See https://github.com/tidepool-org/Loop/pull/180